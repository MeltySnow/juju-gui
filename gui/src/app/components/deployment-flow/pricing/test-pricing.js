/* Copyright (C) 2018 Canonical Ltd. */
'use strict';

const React = require('react');
const enzyme = require('enzyme');

const Analytics = require('test/fake-analytics');
const DeploymentPricing = require('./pricing');

describe('DeploymentPricing', () => {

  const renderComponent = (options = {}) => enzyme.shallow(
    <DeploymentPricing
      addNotification={options.addNotification || sinon.stub()}
      analytics={Analytics}
      applications={options.applications || []}
      changeState={options.addNotification || sinon.stub()}
      charms={options.charms || {}}
      estimate={options.estimate || '3500'}
      generatePath={options.addNotification || sinon.stub()}
      getSLAMachineRates={options.getSLAMachineRates || sinon.stub()}
      listPlansForCharm={options.listPlansForCharm || sinon.stub()}
      machineCount={options.machineCount || '3'}
      setSLA={options.setSLA || sinon.stub()} />
  );

  it('can render', function() {
    const wrapper = renderComponent();
    expect(wrapper).toMatchSnapshot();
  });
});
