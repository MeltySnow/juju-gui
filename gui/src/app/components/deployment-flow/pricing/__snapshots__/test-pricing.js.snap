// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeploymentPricing can render 1`] = `
<div
  className="deployment-pricing"
>
  <DeploymentPlanTable
    addNotification={[Function]}
    applications={Array []}
    charms={Object {}}
    listPlansForCharm={[Function]}
  />
  <div
    className="twelve-col"
  >
    <div
      className="six-col deployment-pricing__secondary-text"
    >
      Monthly cost is determined by price plan, support level, and by usage.
    </div>
    <div
      className="six-col last-col u-align--right deployment-pricing__estimated-price"
    >
      Estimated application cost:
      <span
        className="deployment-pricing__estimated-price-number"
      >
        $
        3500
      </span>
    </div>
  </div>
  <div
    className="twelve-col no-margin-bottom"
    id="support-level"
  >
    <div
      className="six-col deployment-pricing__support-title"
    >
      Choose your level of support
    </div>
    <div
      className="six-col last-col u-align--right deployment-pricing__estimated-price"
    >
      <a
        href="https://www.ubuntu.com/support/plans-and-pricing#ua-support"
        target="_blank"
      >
        Support details
      </a>
    </div>
  </div>
  <DeploymentSupportSelection
    analytics={
      Object {
        "addCategory": [Function],
        "sendEvent": [Function],
      }
    }
    getSLAMachineRates={[Function]}
    machineCount="3"
    setSLA={[Function]}
  />
  <div
    className="twelve-col no-margin-bottom"
  >
    <div
      className="six-col no-margin-bottom deployment-pricing__secondary-text"
    >
      <strong>
        Estimated costs based on 
        3
         
        machines
        .
      </strong>
       Support is billed monthly. You can upgrade the level at any time, and downgrade at the end of a month.
    </div>
    <div
      className="six-col last-col u-align--right no-margin-bottom"
    >
      <Link
        changeState={[Function]}
        classes={Array []}
        clickState={
          Object {
            "gui": Object {
              "deploy": "",
            },
          }
        }
        generatePath={[Function]}
      >
        Unsupported version
      </Link>
    </div>
  </div>
</div>
`;
